MatrixSDK/Crypto/Dehydration/DehydrationService.swift

```diff
@@ -0,0 +1,214 @@
+//
+// Copyright 2023 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import MatrixSDKCrypto
+
+enum DehydrationServiceError: Error {
+    case failedDehydration(Error)
+    case noDehydratedDeviceAvailable(Error)
+    case failedRehydration(Error)
+    case invalidRehydratedDeviceData
+    case failedStoringSecret(Error)
+    case failedRetrievingSecret(Error)
+    case failedRetrievingPrivateKey(Error)
+    case invalidSecretStorageDefaultKeyId
+    case failedRetrievingToDeviceEvents(Error)
+    case failedDeletingDehydratedDevice(Error)
+}
+
+@objcMembers
+public class DehydrationService: NSObject {
+    let deviceDisplayName = "Backup Device"
```

Comment:
Maybe this should be configurable through the SDK settings, for e.g. localisation.

Reply:
Is this customer facing? If so we need to think more on how to name it.